### PR TITLE
consumer(ticdc): add more logs to the consumer

### LINKED ISSUE #11883: kafka consumer add more logs to help debug more easier
Kafka consumer is used for the internal testing purpose, we add more logs into it, to make the daily debug life easier.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #11883 

### What is changed and how it works?

* add more logs when receive messages from the kafka topic
* refactor the code logic about how to handle unordered events.
* only sort events for the canal-json, since unordered events may comes from the partition table. this will be removed after canal-json support outputs table id and partition table id

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Review by CharlesCheung96 — APPROVED** [review]

File: cmd/kafka-consumer/writer.go [hunk 1]
```diff
@@ -317,43 +333,41 @@ func (w *writer) WriteMessage(ctx context.Context, message *kafka.Message) bool
...
-					zap.String("DDL", ddl.Query))
+			// the Query maybe empty if using simple protocol, it's comes from `bootstrap` event, no need to handle it.
+			if ddl.Query == "" {
+				continue
+			}
+
+			if partition == 0 {
+				w.appendDDL(ddl, offset)
 			}
+			needFlush = true
```
Comment (wk989898) [inline cmd/kafka-consumer/writer.go:365]: What happens if not `partition ==0` ?

**Review by wk989898 — APPROVED** [review]

**Pull request merged**